operator/src/main/java/org/projectnessie/operator/reconciler/nessie/dependent/HorizontalPodAutoscalerV2Beta1Dependent.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.operator.reconciler.nessie.dependent;
+
+import io.fabric8.kubernetes.api.model.autoscaling.v2beta1.CrossVersionObjectReferenceBuilder;
+import io.fabric8.kubernetes.api.model.autoscaling.v2beta1.HorizontalPodAutoscaler;
+import io.fabric8.kubernetes.api.model.autoscaling.v2beta1.HorizontalPodAutoscalerBuilder;
+import io.fabric8.kubernetes.api.model.autoscaling.v2beta1.HorizontalPodAutoscalerSpecBuilder;
+import io.fabric8.kubernetes.api.model.autoscaling.v2beta1.MetricSpec;
+import io.fabric8.kubernetes.api.model.autoscaling.v2beta1.MetricSpecBuilder;
+import io.fabric8.kubernetes.api.model.autoscaling.v2beta1.ResourceMetricSourceBuilder;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependent;
+import org.projectnessie.operator.reconciler.KubernetesHelper;
+import org.projectnessie.operator.reconciler.nessie.NessieReconciler;
+import org.projectnessie.operator.reconciler.nessie.resource.Nessie;
+import org.projectnessie.operator.reconciler.nessie.resource.options.AutoscalingOptions;
+
+@KubernetesDependent(labelSelector = NessieReconciler.DEPENDENT_RESOURCES_SELECTOR)
+public class HorizontalPodAutoscalerV2Beta1Dependent
```

Comment:
Difficult because the classes look similar but share no common code in fabric8's code. However we can ask ourselves if it's still useful to support the beta variants, they have been all deprecated for a while now.